cmd/promote/rhobs/rhobs.go

```
@@ -180,52 +181,51 @@ func promoteAllServices(appInterfaceClone *promote.AppInterfaceClone, servicesRe
-		return false, err
+		if targetBlockHasSubscribe(lines, i) {
+			continue
+		}
+		lines[i] = m[1] + newHash
+		updated = true
 	}
+
 	if updated {
-		if err := service.Save(); err != nil {
-			return false, err
+		if err := os.WriteFile(filePath, []byte(strings.Join(lines, "\n")), 0644); err != nil {
```

golangci-lint (2.12.2)

[medium] G306: Expect WriteFile permissions to be 0600 or less

(gosec)